src/osm/FactHandler.cpp

```diff
@@ -238,7 +266,15 @@ void osm2rdf::osm::FactHandler<W>::way(const osm2rdf::osm::Way& way) {
 
         _writer->writeTriple(subj, IRI__RDF_TYPE, IRI__OSM_NODE);
 
-        writeBoostGeometry(subj, IRI__GEOSPARQL__HAS_GEOMETRY, node.geom());
+        if (_config.addGeometryObjects) {
+          const std::string& geomObj = _writer->generateIRI(
+              NAMESPACE__OSM_GEOM, "node_" + std::to_string(node.id()));
+
+          _writer->writeTriple(subj, IRI__GEOSPARQL__HAS_GEOMETRY, geomObj);
+          // node geometry has already been written
```

Comment:
Is this true for all nodes? I think ways can contain untagged nodes.

Reply:
Good catch!